links/src/main/java/juja/microservices/links/service/impl/LinksServiceImpl.java

```diff
@@ -12,19 +15,24 @@
  * @author Ivan Shapovalov
  * @author Vladimir Zadorozhniy
  */
-
+@Slf4j
 @Service
+@AllArgsConstructor
 public class LinksServiceImpl implements LinksService {
-
     private LinksRepository linksRepository;
 
-    public LinksServiceImpl(LinksRepository linksRepository) {
-        this.linksRepository = linksRepository;
-    }
-
     @Override
     public Link saveLink(SaveLinkRequest request) {
-        return null;
+        Link link;
+
+        try {
+            link = linksRepository.getLinkByURL(request.getUrl());
+            log.info("Link already exists {}. ", link.toString());
+        } catch (NotFoundException ex) {
```

Comment:
What is the purpose of throwing Exception in Repo and catching it in Service?
Maybe throw Exception in case of LinkAlreadyExistsException?
That exception have to be thrown if url already exists nd DB and catched by Exception handler.
Then message wrapped and return to slack bot service in response.
Such as 
https://github.com/JujaLabs/teams/blob/a3e5087c2a4c92dd77bf20a43b80469d73faf478/src/main/java/ua/com/juja/microservices/teams/exceptions/ApiExceptionsHandler.java#L45

Reply:
the idea is to save only if the link is unique. 
and throwing an exception instead of return a null object in case link is not found.